app/models/smart_proxy.rb

```diff
@@ -111,15 +111,15 @@ def sanitize_url
   end
 
   def associate_features
-    return true if Rails.env == 'test'
-
     begin
       reply = ProxyAPI::Features.new(:url => url).features
-      if reply.is_a?(Array) && reply.any?
-        self.features = Feature.where(:name => reply.map{|f| Feature.name_map[f]})
+      return errors.add(:base, _('An invalid response was received while requesting available features from this proxy')) unless reply.is_a?(Array)
+      valid_features = reply.map{|f| Feature.name_map[f]}.compact
+      if valid_features.any?
+        self.features = Feature.where(:name => valid_features)
       else
         self.features.clear
-        errors.add :base, _('No features found on this proxy, please make sure you enable at least one feature')
+        errors.add :base, _('No matching features found on this proxy, please make sure you enable at least one feature')
```

Comment:
Not sure what 'matching' means in this context for a user that doesn't know about the bug. Also it tells the user to 'enable at least one feature' but control flow allows Foreman to get here when there was certainly one feature enabled (albeit not known by Foreman)

How about 'Features #{invalid_features} in this proxy are not recognized by Foreman. If these features come from a Smart Proxy plugin, make sure Foreman has the plugin installed too.' when there are invalid features, and leaving the old message "No features found..." for the case where there are no features found?